pandas/core/arrays/datetimes.py

```diff
@@ -418,9 +418,9 @@ def _generate_range(
                 # index is localized datetime64 array -> have to convert
                 # start/end as well to compare
                 if start is not None:
-                    start = start.tz_localize(tz).asm8
+                    start = start.tz_localize(tz, ambiguous).asm8
```

Comment:
Could you also pass `nonexistent` and write a test case for that issue as well?

Reply:
added both